### PR TITLE
ddl: make `TestModifyColumnNullToNotNull` test stable

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve? 
This PR tries to fix a potentially failed assertion that makes TiDB unit test keep hanging and failed because of timeout.

### What is changed and how it works?
The code：
https://github.com/pingcap/tidb/blob/1b72ce5987b3d3396390648cd80788b07b40e4e1/ddl/db_test.go#L2706
in `TestModifyColumnNullToNotNull` has following flaws:

1. Made a write during the schema change of `t1`, in which leads a potential `information schema is changed` error.
2. When the assertion `MustExec` failed, only ddl worker goroutine exits, which leaves ddl job never ends.

This PR removes `MustExec` and does the make outside of `hook`, while changing the time of hook called to avoid `information schema is changed` error.

PS: There are still some other codes that use assertion inside of `hook` in ddl tests, we should refine them all to avoid similar timeout failures, I plan to do the refinements later.

### Check List 

Tests 
 - Unit test
 - Integration test

Code changes
 - NA

Side effects
 - NA

Related changes
 - NA
 
Release note
 - NA

### CODE REVIEW

**Comment by bb7133** [comment]
PTAL @crazycs520 @tiancaiamao @zimulala

File: ddl/db_test.go [hunk 1]
```diff
@@ -2725,18 +2723,18 @@ func (s *testDBSuite1) TestModifyColumnNullToNotNull(c *C) {
 		}
 		c2 := getModifyColumn()
 		if mysql.HasPreventNullInsertFlag(c2.Flag) {
```
Comment (Deardrops) [inline ddl/db_test.go]: Do we need this check? In this line, I think `c2.Flag` is always `HasPreventNullInsertFlag` in this place.

Comment (bb7133) [reply ddl/db_test.go]: You're right, it not necessary.

**Pull request merged**